src/main/java/de/tum/cit/aet/artemis/buildagent/service/SharedQueueProcessingService.java

```diff
@@ -350,6 +393,71 @@ private void processBuild(BuildJobQueueItem buildJob) {
         });
     }
 
+    private void pauseBuildAgent() {
+        if (this.isPaused) {
```

Comment:
```suggestion
        if (isPaused) {
```
You don't use `this` elsewhere in the code